### DISCUSSION TITLE
Possibility to create custom features

### DISCUSSION THREAD

**Discussion by Cifixie** [TanStack/table#4403] [Ideas]
Hi!

I know this is quite advantaged request: At the moment, there isn't much to expand the table-core within it's own eco-system. introducing an idea to append custom features that developers can create independently for their needs.

In example, loading-state - similar API than column visibility - would sit perfectly in this case. In our project we fetch and mesh data from different API's asynchronously. We ended up basically duplicating visibility feature but only externally from Table, adding a new layer, but would in principle fit perfectly within.

**Comment by Pagebakers** [comment]
I'm running into this as well, I'd like to add accessibility features, like focus management.

```
const features = [
  Headers,
  Visibility,
  Ordering,
  Pinning,
  Filters,
  Sorting,
  Grouping,
  Expanding,
  Pagination,
  RowSelection,
  ColumnSizing,
  Focusing, // this is not possible now, but would like to extend the state {focus: [row, cellI]}
] as const
```

Would be great if we can add extra features by passing them to createTable / useReactTable etc, similar to React Table v7.

**Reply by theMattCode** [reply]
Same here. Are there any plans or thoughts about opening the feature API to users? @tannerlinsley Do you eventually need help extending it?
Currently, I kind of misuse the `TableMeta` interface along with some TS hacks injecting own features with access to its state. So far it works great but it definitely has a smell. :smile_cat:

**Comment by SebastianTroc** [comment]
Any thoughts on that @tannerlinsley?
It would be nice to be possible to implement own features, because now that hack https://tanstack.com/table/v8/docs/api/core/column-def#meta with overwriting ColumnMeta types does not work for newer Typescript versions with `@tanstack/table-core` package installed.
